[Bug]: Unknown property 'onBeforeToggle' found (react/no-unknown-property)

### Is there an existing issue for this?

- [x] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Description Overview

this is a follow-up to #3707. the release of React v19 stable means that support for the new `onBeforeToggle` event prop, as well as the camelCased popover attributes `popoverTarget` and `popoverTargetAction`, has now landed in the current latest stable release (they were added [in this PR](https://github.com/facebook/react/pull/27981)).

<!--Show example of your code (as text format), add images/videos/gifs to help explain example-->
<!--and/or Link of repo to where issue is occurring-->

here’s an example trigger + popover:

```tsx
<>
    <button popoverTarget="foo" popoverTargetAction="show">
        Open Popover
    </button>
    <div
        id="foo"
        onBeforeToggle={event => console.log('beforeToggle from', event.oldState, 'to', event.newState)}
        popover="auto"
    >
        <h1>Popover Foo</h1>
    </div>
</>
```

<!--What is happening? / What is the error?-->

example errors:
```
  68:13  error  Unknown property 'onBeforeToggle' found                                          react/no-unknown-property
  78:17  error  Unknown property 'popoverTarget' found, use 'popovertarget' instead              react/no-unknown-property
  79:17  error  Unknown property 'popoverTargetAction' found, use 'popovertargetaction' instead  react/no-unknown-property
```

<!--What command(s) did you run to reproduce issue?-->

these occur when running my `lint` npm run script: `"lint": "eslint .",`


### Expected Behavior

<!--Brief description-->

i expect the new event, `onBeforeToggle` and the camelCased attributes to not be considered “unknown” properties

### eslint-plugin-react version

v7.37.2

### eslint version

v9.16.0

### node version

v22.7.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,11 @@ This change log adheres to standards from [Keep a CHANGELOG](https://keepachange
 
 ## Unreleased
 
+### Fixed
+* [`no-unknown-property`]: support `onBeforeToggle`, `popoverTarget`, `popoverTargetAction` attributes ([#3865][] @acusti)
+
+[#3865]: https://github.com/jsx-eslint/eslint-plugin-react/pull/3865
+
 ## [7.37.3] - 2024.12.23
 
 ### Fixed

--- a/lib/rules/no-unknown-property.js
+++ b/lib/rules/no-unknown-property.js
@@ -259,8 +259,8 @@ const DOM_PROPERTY_NAMES_TWO_WORDS = [
   'onCompositionUpdate', 'onCut', 'onDoubleClick', 'onDrag', 'onDragEnd', 'onDragEnter', 'onDragExit', 'onDragLeave',
   'onError', 'onFocus', 'onInput', 'onKeyDown', 'onKeyPress', 'onKeyUp', 'onLoad', 'onWheel', 'onDragOver',
   'onDragStart', 'onDrop', 'onMouseDown', 'onMouseEnter', 'onMouseLeave', 'onMouseMove', 'onMouseOut', 'onMouseOver',
-  'onMouseUp', 'onPaste', 'onScroll', 'onSelect', 'onSubmit', 'onToggle', 'onTransitionEnd', 'radioGroup', 'readOnly', 'referrerPolicy',
-  'rowSpan', 'srcDoc', 'srcLang', 'srcSet', 'useMap', 'fetchPriority',
+  'onMouseUp', 'onPaste', 'onScroll', 'onSelect', 'onSubmit', 'onBeforeToggle', 'onToggle', 'onTransitionEnd', 'radioGroup',
+  'readOnly', 'referrerPolicy', 'rowSpan', 'srcDoc', 'srcLang', 'srcSet', 'useMap', 'fetchPriority',
   // SVG attributes
   // See https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute
   'crossOrigin', 'accentHeight', 'alignmentBaseline', 'arabicForm', 'attributeName',
@@ -317,9 +317,11 @@ const DOM_PROPERTY_NAMES_TWO_WORDS = [
   'onMouseMoveCapture', 'onMouseOutCapture', 'onMouseOverCapture', 'onMouseUpCapture',
   // Video specific
   'autoPictureInPicture', 'controlsList', 'disablePictureInPicture', 'disableRemotePlayback',
+  // popovers
+  'popoverTarget', 'popoverTargetAction',
 ];
 
-const DOM_PROPERTIES_IGNORE_CASE = ['charset', 'allowFullScreen', 'webkitAllowFullScreen', 'mozAllowFullScreen', 'webkitDirectory'];
+const DOM_PROPERTIES_IGNORE_CASE = ['charset', 'allowFullScreen', 'webkitAllowFullScreen', 'mozAllowFullScreen', 'webkitDirectory', 'popoverTarget', 'popoverTargetAction'];
 
 const ARIA_PROPERTIES = [
   // See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes

--- a/tests/lib/rules/no-unknown-property.js
+++ b/tests/lib/rules/no-unknown-property.js
@@ -188,6 +188,15 @@ ruleTester.run('no-unknown-property', rule, {
         </div>
       `,
     },
+    {
+      code: `
+        <div>
+          <button popoverTarget="my-popover" popoverTargetAction="toggle">Open Popover</button>
+
+          <div id="my-popover" onBeforeToggle={this.onBeforeToggle} popover>Greetings, one and all!</div>
+        </div>
+      `,
+    },
   ]),
   invalid: parsers.all([
     {

